Quantum-Control-Applications/Superconducting/Single-Flux-Tunable-Transmon/17_cryoscope_1ns.py

```diff
@@ -270,18 +276,25 @@ def baked_waveform(waveform, pulse_duration):
             # Convert the results into Volts
             I, Q = u.demod2volts(I, readout_len), u.demod2volts(Q, readout_len)
             # Bloch vector Sx + iSy
-            qubit_state = (state[:, 0] * 2 - 1) + 1j * (state[:, 1] * 2 - 1)
+            Sx = (state[:, 0] * 2 - 1) - np.mean(
+                (state[:, 0] * 2 - 1)[zeros_before_pulse + 10 : len(flux_waveform) - zeros_after_pulse - 10]
+            )
+            Sy = (state[:, 1] * 2 - 1) - np.mean(
+                (state[:, 1] * 2 - 1)[zeros_before_pulse + 10 : len(flux_waveform) - zeros_after_pulse - 10]
+            )
+            qubit_state = Sx + 1j * Sy
 
         # Progress bar
         progress_counter(iteration, n_avg, start_time=results.get_start_time())
         # Accumulated phase: angle between Sx and Sy
         qubit_phase = np.unwrap(np.angle(qubit_state))
-        qubit_phase = qubit_phase - qubit_phase[-1]
         # Filtering and derivative of the phase to get the averaged frequency
-        detuning = signal.savgol_filter(qubit_phase / 2 / np.pi, 13, 3, deriv=1, delta=0.001)
+        detuning = signal.savgol_filter(qubit_phase / 2 / np.pi, 3, 2, deriv=1, delta=1)
         # Flux line step response in freq domain and voltage domain
-        step_response_freq = detuning / np.average(detuning[-int(const_flux_len / 2) :])
-        step_response_volt = np.sqrt(step_response_freq)
+        step_response_freq = detuning / np.average(
+            detuning[len(flux_waveform) - zeros_after_pulse - 110 : len(flux_waveform) - zeros_after_pulse - 10]
+        )
```

Comment:
I don't like the hardcoded 10 because it depends on the time constant of the exponential....
Why not keeping `np.average(detuning[-int(const_flux_len / 2) :])`?